### DISCUSSION TITLE
[Feature] mobile app: share to app

### DISCUSSION THREAD

**Discussion by mtettke123** [immich-app/immich#2659] [Feature Request]
### The feature

I'm using the mobile app to backup photos from my phone, which works great.

From time to time I get a photo sent by someone vis messenger or mail that I would like to add to immich. I could probably store it in my camera folder and it would get uploaded, but it would be much nicer of it would be possible to use the app as share-to for images. This way, I could open the images in the message, click share to immich and it would be uploaded. Great, easy workflow.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by Huber1** [comment]
This is a feature I would also really like to see implemented.
As far as I can tell, the app needs to register itself for being shown in the share dialog. On android this can be done with an intent-filter. On ios there should be something similar.
When sharing, the os is calling native code. This native code must then trigger the upload mechanism of immich

**Comment by OverHash** [comment]
#7746 (a duplicate of this request) shows how this would look like on native UI on Android, and also describes some of the required intents to be implemented.

**Comment by ResponsiblyUnranked** [comment]
Yep I agree this is definitely something I'd appreciate! Being able to upload specific photos from other apps, without having to add the folder as a 'backup' folder would be nice.

**Comment by Fmstrat** [comment]
A huge advantage of this feature is when people send you photos. Get something on Signal? Add it to your Immich.

**Comment by spark108** [comment]
Yes, a very useful function. Very often they send something, I download it from somewhere and it is not always convenient to go into the application and add each new album or manually upload from existing ones. Of course, you can make an automatic upload, but I don’t want all the images to end up in the cloud without going through them.

Yesterday my brother wrote, said that it is inconvenient for his wife to constantly go in and add something.

But there are cases when it is simply difficult to explain to a person, he is used to clicking “Share” and forgetting.

Thank you, I would like to see this feature sooner.

-----

Да, очень нужная функция. Очень часто присылают что-то, скачиваю откуда-то и не всегда удобно заходить в приложение и добавлять каждый новый альбом или вручную выгружать из существующих. Можно конечно сделать автоматическую выгрузку, но не хочется что бы все изображения без перебора попадали в облако.

Вчера еще брат написал, сказал, что его жене неудобно постоянно заходить и что-то добавлять.

А ведь есть случаи когда человеку просто сложно объяснить, он привык нажать "Поделится" и забыть.

Спасибо, хотелось бы побыстрее увидеть эту возможность.

**Comment by CarlvO** [comment]
Plus 1 from me. I don't want to do auto-backup of all my images. Lots of images I make are one-time use only. So I prefer selecting them in the gallery and then share them to my photo service. It's possible to upload selected images from the web-client, but it's more cumbersome that way.

**Comment by spawnpeekboi** [comment]
and I can back up photo in my hidden album in IOS system without unhiding them
the current icloud takesout work like shit, the metadata is a mess

**Comment by MrDiamond10** [comment]
I agree as well. This feature would be great. I created one folder in my android and I am manually saving images from other sources to this folder but it is not really convenient. So +1 from me :) hope soon

**Comment by sasha-id** [comment]
+1

**Comment by MSSandroid** [comment]
This flutter plugin seems to enable the functionality https://pub.dev/packages/receive_sharing_intent . The Readme already has a good explanation. 
I am not familiar with flutter or dart but can try to implement the feature.
After sharing the photo with immich, the user should probably select an album in which to upload. 

Any ideas about how to implement the upload/addition of the photos themselves most "efficiently" or objections against using the mentioned flutter plugin? I plan to implement the feature in the next weeks if its not tackled by somebody else already.

**Comment by vogtrj** [comment]
This would be fantastic! Doubly annoying is that the Google Photos app doesn't have a "download to device" feature but it does have app-to-app sharing. People often share photos to me through G Photos, and it's a hassle to get the photo from G Photos to Immich. From Google Photos, I have to share the image to some other arbitrary intermediary app which DOES let me download the photo to my phone, then from that intermediary app download the photo to a folder that Immich has been configured to back up. Direct share from Google Photos to Immich as a means to upload the file would be SO much easier!

**Reply by imthefrizzlefry** [reply]
Google Photos does have a download option, and Immich does good at backing up those photos just fine for me.

**Comment by josx2** [comment]
This is highly needed IMO..... being able to share an asset to an app on the phone is much needed.

**Comment by Cromoray** [comment]
+1 for me, sometimes i just want to some specific photo of some album to be backupped so today i tried to share from gallery but didn't found the app in the list.
I'm waiting for this feature so much.

**Comment by bo0tzz** [comment]
This is now implemented.

**Comment by HoroTW** [comment]
This broke for me recently - does it still work for others?
(Using up to date immich server and the android app)

**Comment by josx2** [comment]
Yeah, broke for me too with the latest version.  I cut over to beta
timeline as well
